CMakeLists.txt

```diff
@@ -130,6 +141,43 @@ endforeach(func_name __PRETTY_FUNCTION__ __FUNCTION__)
 # Libraries
 #
 
+# CUDA compiler
+if(WITH_CUDA)
+  set(CMAKE_CUDA_STANDARD ${CMAKE_CXX_STANDARD})
+  set(MINIMAL_CUDA_VERSION 9.0)
+  option_enum(varname "WITH_CUDA_COMPILER" help_text "CUDA compiler"
+              default_value "nvcc" possible_values "nvcc;hip;clang")
+  if(WITH_CUDA_COMPILER STREQUAL "nvcc")
+    find_package(CUDA ${MINIMAL_CUDA_VERSION} REQUIRED)
+    find_package(CUDACompilerNVCC ${MINIMAL_CUDA_VERSION} REQUIRED)
+  elseif(WITH_CUDA_COMPILER STREQUAL "hip")
+    set(ROCM_HOME "/opt/rocm" CACHE FILEPATH "Path to AMD ROCm")
+    list(APPEND CMAKE_MODULE_PATH "${ROCM_HOME}/hip/cmake")
+    find_package(HIP 1.5.18494 MODULE REQUIRED)
+    # patch HCC_PATH environment variable and reload HIP
+    if(HIP_VERSION VERSION_LESS "3.1")
+      set(HCC_PATH "${HIP_ROOT_DIR}")
+    else()
+      set(HCC_PATH "${ROCM_HOME}/hcc")
+    endif()
+    find_package(HIP 1.5.18494 MODULE REQUIRED)
+    find_package(CUDACompilerHIP 6.0 REQUIRED)
```

Comment:
Where does that `6.0` version number come from? Is that the Clang version of hipcc? I don't think it makes sense to require a minimum version there because the HIP version already implies a certain minimum Clang version.

Reply:
It's the Clang version. I put 6.0 here since this is the oldest Clang version we're testing. Do you know where I could find a table showing the HIP <=> Clang version mapping?

Reply:
I've removed that Clang check in be61c3a1e.